src/account_manager.rs

```diff
@@ -259,62 +330,65 @@ impl AccountManager {
     }
 
     /// Gets the account associated with the given identifier.
-    pub fn get_account(&self, account_id: &AccountIdentifier) -> crate::Result<Account> {
-        let mut account = crate::storage::get_account(&self.storage_path, &account_id)?;
-        account.set_storage_path(self.storage_path.clone());
-        Ok(account)
+    pub fn get_account(&self, account_id: &AccountIdentifier) -> crate::Result<AccountGuard> {
+        let accounts = self.accounts.read().unwrap();
+        accounts
+            .get(account_id)
+            .cloned()
+            .ok_or(crate::WalletError::AccountNotFound)
     }
 
     /// Gets the account associated with the given alias (case insensitive).
-    pub fn get_account_by_alias<S: Into<String>>(&self, alias: S) -> Option<Account> {
+    pub fn get_account_by_alias<S: Into<String>>(&self, alias: S) -> Option<AccountGuard> {
         let alias = alias.into().to_lowercase();
-        if let Ok(accounts) = self.get_accounts() {
-            accounts.into_iter().find(|acc| acc.alias().to_lowercase() == alias)
-        } else {
-            None
-        }
+        self.get_accounts().into_iter().find(|acc| {
```

Comment:
This method invokes `get_accounts` which takes the entire account `HashMap`, iterates through the elements, clones them, collects them into a vector, and then searches that vector for the required element by cloning the alias string into a new allocation, converting it to lowercase, and then comparing it to the desired string. This comes with a very significant amount of unnecessary overhead. If you must use linear search (an alias -> account mapping would be better) then I suggest something like this instead:

```rust
self.accounts.read().unwrap().values().find(|a| {
    // alias() should return a &str rather than a new string
    a.alias().chars().zip(alias.chars()).all(|(x, y)| x.to_lowercase() == y)
})
```